server/src/main/java/org/apache/druid/segment/realtime/appenderator/StreamAppenderator.java

```diff
@@ -1098,16 +1112,16 @@ SinkSchemaAnnouncer getSinkSchemaAnnouncer()
   /**
    * Unannounces the given base segment and all its upgraded versions.
    */
-  private void unannounceAllVersionsOfSegment(DataSegment baseSegment) throws IOException
+  private void unannounceAllVersionsOfSegment(DataSegment baseSegment)
```

Comment:
Maybe add a comment in the javadoc that this method should be synchronized on the corresponding sink. Alternatively, you could even take the sink as an argument and synchronize on it inside the method rather than relying on the callers to do so.

Reply:
Synchronized inside the method. Thanks!